src/core/namespace.c

```diff
@@ -2097,6 +2097,7 @@ int setup_namespace(const NamespaceParameters *p, char **error_path) {
                 DISSECT_IMAGE_FSCK |
                 DISSECT_IMAGE_USR_NO_ROOT |
                 DISSECT_IMAGE_GROWFS |
+                DISSECT_IMAGE_MKDIR_MOUNTPOINTS |
```

Comment:
what's the usecase for this? when would we ever not set this?

if people want immutable images they should set the r/o flag

Reply:
As explained in the commit messages, dissect is an introspecting tool and as such it should not modify the introspected images by default. If anything is missing to properly mount the image it should complain and exit  (regardless of whether if it's mounted ro or rw).

If modifying the image is wanted then passing a specific option to allow that (`--mkdir-mountpoints` or such) seems a better approach.

Reply:
but the modes where it just introspects the image all set the r/o flag, hence the mkdir doesn't work anyway? so why is any new flag required here?

Reply:
See the use case I provided below with `--mount`. This mode, by default, modifies the image to create /efi if it doesn't exist. You will argue that `--read-only` could be passed but the fact that the user wants to mount the image RW doesn't mean that the tool is allowed to do various image modifications. And the man page doesn't say anything about that.